internal/settings/providers/namecom/createrecord.go

```diff
@@ -0,0 +1,66 @@
+package namecom
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+
+	"github.com/qdm12/ddns-updater/internal/settings/constants"
+	"github.com/qdm12/ddns-updater/internal/settings/errors"
+	"github.com/qdm12/ddns-updater/internal/settings/utils"
+)
+
+func (p *Provider) createRecord(ctx context.Context, client *http.Client,
+	ip net.IP) (err error) {
+	recordType := constants.A
+	if ip.To4() == nil {
+		recordType = constants.AAAA
+	}
+
+	u := &url.URL{
+		Scheme: "https",
+		Host:   "api.name.com",
+		Path:   fmt.Sprintf("/v4/domains/%s/records", p.domain),
+		User:   url.UserPassword(p.username, p.password),
+	}
+
+	postRecordsParams := struct {
+		Host   string `json:"host"`
+		Type   string `json:"type"`
+		Answer string `json:"answer"`
+		TTL    int    `json:"ttl"`
+	}{
+		Host:   p.host,
+		Type:   recordType,
+		Answer: ip.String(),
+		TTL:    p.ttl,
+	}
+
+	bodyBytes, err := json.Marshal(postRecordsParams)
+	if err != nil {
+		return fmt.Errorf("%w: %w", errors.ErrRequestMarshal, err)
+	}
+
+	request, err := http.NewRequestWithContext(ctx, http.MethodPost, u.String(), bytes.NewBuffer(bodyBytes))
+	if err != nil {
+		return fmt.Errorf("%w: %w", errors.ErrBadRequest, err)
+	}
```

Comment:
> Any API calls requiring the use of the HTTP POST or PUT methods (for example, [Search](https://www.name.com/Domains#Search) or [UpdateEmailForwarding](https://www.name.com/EmailForwardings#UpdateEmailForwarding)) must set the Content-Type header to application/json.

Let's set that header on the request, and also set user agent headers.

For example other providers use a `setHeaders` method (or it can be a function):

```go
func (p *Provider) setHeaders(request *http.Request) {
	headers.SetContentType(request, "application/json")
	headers.SetAccept(request, "application/json")
	headers.SetUserAgent(request)
}
```